app/services/reports/monthly_bulk_report_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Reports
+  class MonthlyBulkReportService < ::WasteExemptionsEngine::BaseService
+    def run(first_day_of_the_month)
+      @first_day_of_the_month = first_day_of_the_month
+
+      # Generate report
+      temp_file.write(bulk_report)
+
+      # upload file to s3
+
+      # rubocop:disable Layout/CommentIndentation
+      # if upload successfull
+        # record content created
+      # else
+        # retry 3 times then exit with failure
+      # end
+      # rubocop:enable Layout/CommentIndentation
+    ensure
+      temp_file.close
+      temp_file.unlink
+    end
+
+    private
+
+    def temp_file
+      @_temp_file ||= Tempfile.new(file_name)
+    end
+
+    def file_name
+      data_from_date = @first_day_of_the_month.strftime("%Y%m%d")
```

Comment:
Would prefer to stick with the convention of placing all date formatters in `config/initializers/time_formats.rb` and then referring to it here using `to_formatted_s(:my_format)` rather than using `strftime()` directly.

Reply:
Yes, I have done that for the dates in the presenter but this one didn't look "important" enough? I mean, is not a date format we will reuse elsewhere any time soon. Perhaps I can give it a name that includes the `reports` key :)